Canary: Apply single font sizes as overrides and remove font family definition on headings

mikachan opened the pull request:
This changes the way the single font size preset is applied in Canary, as it seems to be causing issues with how typography is used in Global Styles.

Instead of overwriting each font size preset with the same value, this overrides each font size setting to the same preset (`small`).

This also removes the font family setting for `core/heading`, as this doesn't seem to be required anymore and it makes managing the font families from Global Styles less confusing.

Fixes https://github.com/WordPress/twentytwentythree/issues/278.

carolinan commented on styles/canary.json:
```diff
@@ -52,19 +52,20 @@
 				},
 				{
-					"size": "0.75rem",
+					"size": "1.125rem",
```
If the only change to the fontSizes compared to theme.json is that they are not fluid, I wonder if this could be simplified 🤔

mikachan replied:
The difference with the Canary fontSizes is that they're all the same size.

mikachan commented:
I'm going to bring this in as I believe it addresses the issues from https://github.com/WordPress/twentytwentythree/issues/278, but please open any further related issues if needed.
